packages/datadog-plugin-rhea/src/index.js

```diff
@@ -0,0 +1,215 @@
+'use strict'
+
+const dd = Symbol('datadog')
+const circularBufferConstructor = Symbol('circularBufferConstructor')
+
+function createWrapSend (tracer, config, instrumenter) {
+  return function wrapSend (send) {
+    return function sendWithTrace (msg, tag, format) {
+      patchCircularBuffer(this, instrumenter)
+      const name = this.options && this.options.target && this.options.target.address
+        ? this.options.target.address : 'amq.topic'
+      const { host, port } = getHostAndPort(this.connection)
+      return tracer.trace('amqp.send', {
+        tags: {
+          'component': 'rhea',
+          'resource.name': name,
+          'service.name': config.service || `${tracer._service}-amqp-producer`,
+          'span.kind': 'producer',
+          'amqp.link.target.address': name,
+          'amqp.link.role': 'sender',
+          'out.host': host,
+          'out.port': port
+        }
+      }, (span, done) => {
+        addDeliveryAnnotations(msg, tracer, span)
+        const delivery = send.apply(this, arguments)
+        delivery[dd] = { done, span }
+        return delivery
+      })
+    }
+  }
+}
+
+function createWrapConnectionDispatch (tracer, config) {
+  return function wrapDispatch (dispatch) {
+    return function dispatchWithTrace (eventName, obj) {
+      if (eventName === 'disconnected' && this.local_channel_map) {
+        const error = obj.error || this.saved_error
+        for (const key in this.local_channel_map) {
+          const session = this.local_channel_map[key]
+          if (session) {
+            finishDeliverySpans(session.incoming, error)
+            finishDeliverySpans(session.outgoing, error)
+          }
+        }
+      }
+      return dispatch.apply(this, arguments)
+    }
+  }
+}
+
+function createWrapReceiverDispatch (tracer, config, instrumenter) {
+  return function wrapDispatch (dispatch) {
+    return function dispatchWithTrace (eventName, msgObj) {
+      patchCircularBuffer(this, instrumenter)
+      if (eventName === 'message' && msgObj) {
+        const options = msgObj.receiver && msgObj.receiver.options ?
+          msgObj.receiver.options : {}
+        const name = options.source && options.source.address
+          ? options.source.address : 'amq.topic'
+        const childOf = msgObj.message ?
```

Comment:
Using functions for each of the above variables would make the code easier to read.

Reply:
The same could also be applied to the other wrappers.